docs/src/pages/components/text-fields/text-fields.md

```diff
@@ -135,6 +135,44 @@ Below is an example using the [`InputBase`](/api/input-base/) component, inspire
 
 🎨 If you are looking for inspiration, you can check [MUI Treasury's customization examples](https://mui-treasury.com/styles/text-field).
 
+## `useFormControl`
+
+For advanced customization use cases, a `useFormControl()` hook is exposed.
+It returns the context value of the parent form control.
+The `FormLabel`, `FormHelperText`, `Input`, `InputLabel` components use this hook internally.
+
+#### API
+
+```jsx
+import { useFormControl } from '@material-ui/core/FormControl';
+```
+
+#### Returns
+
+`value` (_Object_):
+
+- `value.adornedStart` (_Bool_): Indicate whether the child `Input` or `Select` component has a start adornment.
+- `value.setAdornedStart` (_Func_): Setter function for `adornedStart` state value.
+- `value.color` (_String_): The theme color is being used, inherited from `FormControl` `color` prop .
+- `value.disabled` (_Bool_): Indicate whether the component is being displayed in a disabled state, inherited from `FormControl` `disabled` prop.
+- `value.error` (_Bool_): Indicate whether the component is being displayed in an error state, inherited from `FormControl` `error` prop
+- `value.filled` (_Bool_): Indicate whether the child `Input` or `Select` component is being filled
```

Comment:
```suggestion
- `value.filled` (_Bool_): Indicate whether input is filled
```
One can use this with a custom implementation of `Input` and `Select` so we should be more generic here. Otherwise we don't need to expose `onFilled`.